go/pkg/router/dataplane.go

```diff
@@ -698,24 +695,47 @@ func (p *scionPacketProcessor) processPkt(rawPkt []byte,
 		}
 		return p.processOHP()
 	case scion.PathType:
+		res, err := p.processSCION()
+
+		if err != nil {
+			return processResult{}, err
+		}
+
 		address := p.scionLayer.SrcIA
-		stringAddress := address.String()
+		identifier := fmt.Sprintf("%s-%d", address.String(), res.EgressID)
```

Comment:
And why not just use your buildIdentifier() function here?

Reply:
I put a dash because I didn't know if the address had a fixed length or not. I didn't have access to the buildIdentifier function there, I'm moving the function to the dataplane.

Reply:
Indeed, for the string representation the address is probably not fixed length.
A more elegant solution would be to use a map from [x]byte instead of string, where x is the number of bytes required for the address plus the egress ID.
A map based on a byte array is significantly faster than a map based on a string array.

Reply:
Actually I already tried that for the rate limiter and the go language doesn't allow the map to have byte arrays as key type

Reply:
I guess you tried with a slice, not with an array.
With an array it works, e.g.: `m := make(map[[12]byte]int)`